tests/x509/verification/test_limbo.py

```diff
@@ -131,12 +127,33 @@ def _limbo_testcase(id_, testcase):
     if max_chain_depth is not None:
         builder = builder.max_chain_depth(max_chain_depth)
 
-    verifier = builder.build_server_verifier(peer_name)
+    verifier: ServerVerifier | ClientVerifier
+    if testcase["validation_kind"] == "SERVER":
+        assert testcase["extended_key_usage"] == [] or testcase[
+            "extended_key_usage"
+        ] == ["serverAuth"]
+        peer_name = _get_limbo_peer(testcase["expected_peer_name"])
+        verifier = builder.build_server_verifier(peer_name)
+    else:
+        assert testcase["extended_key_usage"] == ["clientAuth"]
+        verifier = builder.build_client_verifier()
 
     if should_pass:
-        built_chain = verifier.verify(
-            peer_certificate, untrusted_intermediates
-        )
+        if isinstance(verifier, ServerVerifier):
+            built_chain = verifier.verify(
+                peer_certificate, untrusted_intermediates
+            )
+        else:
+            verified_client = verifier.verify(
+                peer_certificate, untrusted_intermediates
+            )
+
+            expected_subjects = {
+                _get_limbo_peer(p) for p in testcase["expected_peer_names"]
+            }
+            assert expected_subjects.issubset(verified_client.subjects)
```

Comment:
Why is this a subset check and not an equal check?

Reply:
My original thinking for that was to enable simpler tests on the limbo side: if we know that just a single particular subject must appear and don't care about the rest, all we need is a subset check. Thinking about it more though, having it be exact probably makes the tests more descriptive.